src/TriggerBinding/SqlTriggerTargetScaler.cs

```diff
@@ -14,24 +17,74 @@ internal sealed class SqlTriggerTargetScaler : ITargetScaler
+            this._userTable = userTable ?? throw new ArgumentNullException(nameof(userTable));
         }
 
         public TargetScalerDescriptor TargetScalerDescriptor { get; }
 
         public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
         {
-            SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+            try
+            {
+                SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+
+                // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
+                int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
 
-            // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
-            int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
+                return GetScaleResultInternal(concurrency, metrics.UnprocessedChangeCount);
+            }
+            catch (Exception ex)
+            {
+                // If the exception is SQL exception and indicates that the object name is invalid, it means that the global state and leases table are not created
+                // Check for the error number 208 https://learn.microsoft.com/en-us/sql/relational-databases/errors-events/mssqlserver-208-database-engine-error?view=sql-server-ver17
+                if (ex is SqlException sqlEx && sqlEx.Number == 208)
+                {
+                    // Check if we already tried to spin up the worker that starts the listener which will create those tables.
+                    // If we have already tried, we will return 0 as the target worker count.
+                    this._logger.LogWarning("Invalid object name detected. SQL trigger tables not found.");
+                    if (_firstTableCreationWarmupAttempt != DateTime.MinValue && DateTime.UtcNow - _firstTableCreationWarmupAttempt > TimeSpan.FromMinutes(2))
+                    {
+                        // If we have already checked within the last 2 minutes, we will return 0 as the target worker count.
+                        this._logger.LogWarning("Returning 0 as the target worker count since the GetMetrics query threw an 'Invalid object name detected' error and we've exceeded the warmup period for scaling up a new instance to create the required state tables.");
+                        return new TargetScalerResult
+                        {
+                            TargetWorkerCount = 0
+                        };
+                    }
+                    else
+                    {
+                        // Check if there are any changes in the user table.
```

Comment:
```suggestion
                        // Check if there are any changes in the user table. Since we don't have a leases table that means
                        // we haven't processed any of the changes yet so we can just check if there's any changes
                        // for the table at all (no last sync point)
```